### DISCUSSION TITLE
Too much time spent on postprocess

### DISCUSSION THREAD

**Discussion by GraftonYang** [ultralytics/ultralytics#8516] [Q&A]
I used an onnx model which was pretrained by yolov8n.pt. When i use the onnx model detect a picture which is 1685*2041, the postprocess is about 5 seconds, but the preprocess and inference are very low. How can i resolve the problem?🙏

**Comment by pderrenger** [comment]
Hello! It sounds like the post-processing step is taking unexpectedly long for your ONNX model derived from `yolov8n.pt`. Here are a couple of suggestions that might help reduce the time spent on post-processing:

1. **Optimize NMS**: The Non-Maximum Suppression (NMS) step can be a bottleneck. Consider using a more efficient NMS implementation if available in your processing environment.

2. **Reduce Output**: If your use case allows, reducing the number of classes or adjusting the confidence threshold (`conf`) might lead to fewer detections and, consequently, faster post-processing.

3. **Batch Processing**: If you're processing images individually, try batch processing if your setup allows. This can sometimes reduce overheads associated with post-processing.

4. **ONNX Optimization Tools**: Utilize ONNX optimization tools that might help streamline the model further, potentially impacting post-processing positively.

Here's a quick example of adjusting the confidence threshold when using the model, which might help reduce the load on post-processing:

```python
# Assuming you're using a Python wrapper for ONNX inference
results = model.predict(image, conf_threshold=0.5)
```

If these suggestions don't help, it might be worth profiling the post-processing step to identify the exact bottleneck. 🕵️‍♂️

Let us know how it goes!
